modules/aggregation-layers/src/heatmap-layer/triangle-layer.ts

```diff
@@ -33,61 +32,73 @@ type _TriangleLayerProps = {
   vertexCount: number;
   colorTexture: Texture;
   maxTexture: Texture;
-  texture: Texture;
+  weightsTexture: Texture;
 };
 
 export default class TriangleLayer extends Layer<_TriangleLayerProps> {
   static layerName = 'TriangleLayer';
 
   state!: {
     model: Model;
+    positions: Buffer,
+    texCoords: Buffer,
   };
 
   getShaders() {
     return {vs, fs, modules: [project32]};
   }
 
   initializeState({device}: LayerContext): void {
+    console.log('TriangleLayer#initializeState()'); // TODO(donmccurdy): DO NOT SUBMIT.
+    this.setState({model: this._getModel(device)});
+  }
+
+  _getModel(device: Device): Model {
+    console.log('TriangleLayer#_getModel()'); // TODO(donmccurdy): DO NOT SUBMIT.
+
     const attributeManager = this.getAttributeManager()!;
+    const {vertexCount, data} = this.props;
+
+    // TODO(donmccurdy): This is probably doing nothing unless passed to the Model?
     attributeManager.add({
       positions: {size: 3, noAlloc: true},
       texCoords: {size: 2, noAlloc: true}
     });
-    this.setState({
-      model: this._getModel(device)
-    });
-  }
 
-  _getModel(device: Device): Model {
-    const {vertexCount} = this.props;
-
-    // @ts-expect-error TODO(v9): Likely a real error.
+    // TODO(v9): Noisy compiler errors about uniforms here.
     return new Model(device, {
       ...this.getShaders(),
       id: this.props.id,
-      geometry: new Geometry({
-        topology: 'triangle-fan-webgl',
-        vertexCount
-      })
+      attributes: data.attributes,
+      bufferLayout: [
+        {name: 'positions', format: 'float32'},
+        {name: 'texCoords', format: 'float32'},
+      ],
+      topology: 'triangle-fan-webgl',
```

Comment:
TODO: Compare to changes in #8307. I don't think that approach works without `BufferWithAccessor`, but there could be other things I've missed here.